xrpl-wasm-stdlib/src/core/ledger_objects/mod.rs

```diff
@@ -340,59 +283,47 @@ impl FieldGetter for Amount {
     #[inline]
     fn get_from_current_ledger_obj(field_code: i32) -> Result<Self> {
         const BUFFER_SIZE: usize = 48;
-        let mut buffer = core::mem::MaybeUninit::<[u8; BUFFER_SIZE]>::uninit();
-        let result_code = unsafe {
-            get_current_ledger_obj_field(field_code, buffer.as_mut_ptr().cast(), BUFFER_SIZE)
-        };
-        match_result_code(result_code, || {
-            Amount::from(unsafe { buffer.assume_init() })
-        })
+        match get_variable_size_field::<BUFFER_SIZE, _>(field_code, |fc, buf, size| unsafe {
+            get_current_ledger_obj_field(fc, buf, size)
+        }) {
+            Result::Ok((buffer, _len)) => Result::Ok(Amount::from(buffer)),
+            Result::Err(e) => Result::Err(e),
+        }
     }
 
     #[inline]
     fn get_from_current_ledger_obj_optional(field_code: i32) -> Result<Option<Self>> {
         const BUFFER_SIZE: usize = 48;
-        let mut buffer = core::mem::MaybeUninit::<[u8; BUFFER_SIZE]>::uninit();
-        let result_code = unsafe {
-            get_current_ledger_obj_field(field_code, buffer.as_mut_ptr().cast(), BUFFER_SIZE)
-        };
-        match_result_code_optional(result_code, || {
-            Some(Amount::from(unsafe { buffer.assume_init() }))
-        })
+        match get_variable_size_field_optional::<BUFFER_SIZE, _>(
+            field_code,
+            |fc, buf, size| unsafe { get_current_ledger_obj_field(fc, buf, size) },
+        ) {
+            Result::Ok(opt) => Result::Ok(opt.map(|(buffer, _len)| Amount::from(buffer))),
+            Result::Err(e) => Result::Err(e),
+        }
     }
 
     #[inline]
     fn get_from_ledger_obj(register_num: i32, field_code: i32) -> Result<Self> {
         const BUFFER_SIZE: usize = 48;
-        let mut buffer = core::mem::MaybeUninit::<[u8; BUFFER_SIZE]>::uninit();
-        let result_code = unsafe {
-            get_ledger_obj_field(
-                register_num,
-                field_code,
-                buffer.as_mut_ptr().cast(),
-                BUFFER_SIZE,
-            )
-        };
-        match_result_code(result_code, || {
-            Amount::from(unsafe { buffer.assume_init() })
-        })
+        match get_variable_size_field::<BUFFER_SIZE, _>(field_code, |fc, buf, size| unsafe {
```

Comment:
Minor nit: Feels like `BUFFER_SIZE` should be pulled out to something (maybe in Amount?), like `AMOUNT_SIZE`?

See `HASH128_SIZE` for an example.